packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -272,6 +274,61 @@ class ErrorHandlingFile
     );
   }
 
+  /// This copy method attempts to handle file system errors from both reading
+  /// and writing the copied file.
+  @override
+  File copySync(String newPath) {
+    final File resultFile = fileSystem.file(newPath);
+    // First check if the source file can be read. If not, bail through error
+    // handling.
+    _runSync<void>(
+      () => delegate.openSync(mode: FileMode.read).closeSync(),
+      platform: _platform,
+      failureMessage: 'Flutter failed to copy $path to $newPath due to source location error'
+    );
+    // Next check if the destination file can be written. If not, bail through
+    // error handling.
+    _runSync<void>(
+      ()  {
+        resultFile.createSync(recursive: true);
+        resultFile.openSync(mode: FileMode.writeOnly).closeSync();
+      },
+      platform: _platform,
+      failureMessage: 'Flutter faiuled to copy $path to $newPath due to destination location error'
+    );
+    // If both of the above checks passed, attempt to copy the file and catch
+    // any thrown errors.
+    try {
+      return wrapFile(delegate.copySync(newPath));
+    } on FileSystemException {
+      // Proceed below
+    }
+    // If the copy failed but both of the above checks passed, copy the bytes
+    // directly.
+    _runSync(() {
+      final RandomAccessFile source = delegate.openSync(mode: FileMode.read);
+      final RandomAccessFile sink = resultFile.openSync(mode: FileMode.writeOnly);
+      try {
+        // 64k is the same sized buffer used by dart:io for `File.openRead`.
+        final Uint8List buffer = Uint8List(64 * 1024);
+        final int totalBytes = source.lengthSync();
+        int bytes = 0;
+        while (bytes < totalBytes) {
+          final int chunkLength = source.readIntoSync(buffer);
+          sink.writeFromSync(buffer, 0, chunkLength);
+          bytes += chunkLength;
+        }
+      } finally {
+        source?.closeSync();
+        sink?.closeSync();
+      }
+    }, platform: _platform, failureMessage: 'Flutter failed to copy $path to $newPath due to unknown error');
+    // The original copy failed, but the manual copy worked. Report an analytics event to
+    // track this to determine if this code path is actually hit.
+    ErrorHandlingEvent('copy-fallback').send();
```

Comment:
This will give us the count of `copySync` failing, but the manual copy succeeding. Would any other information be helpful?

Reply:
If one of the file open checks fails, it is likely to be a regular old permission issue. If not, crash logging will handle it.

if the fallback copySync fails, assuming it is not one of the known issues, crash logging will handle that too